pyproject.toml

```diff
@@ -1,41 +1,58 @@
-[project]
+[project.optional-dependencies]
+dev = [
+    "pytest",
+    "pylint",
+]
+ngsolve = [
+    "ngsolve",
+]
+firedrake = [
+    "firedrake",
+]
+
+[tool.poetry]
 name = "ngsPETSc"
-version = "0.0.5"
-description = "NGSolve/Netgen interface to PETSc."
-readme = "README.md"
+version = "0.0.6"
+description = "NGSolve/Netgen interface to PETSc"
 authors = [
-  {name = "Umberto Zerbinati", email = "umberto.zerbinati@maths.ox.ac.uk"},
-  {name = "Patrick E. Farrell", email = "patrick.farrell@maths.ox.ac.uk"},
-  {name = "Stefano Zampini", email = "stefano.zampini@kaust.edu.sa"},
-  {name = "Jack Betteridge", email = "J.Betteridge@imperial.ac.uk"},
+  "Umberto Zerbinati <umberto.zerbinati@maths.ox.ac.uk>",
+  "Patrick E. Farrell <patrick.farrell@maths.ox.ac.uk>",
+  "Stefano Zampini <stefano.zampini@kaust.edu.sa>",
+  "Jack Betteridge <J.Betteridge@imperial.ac.uk>",
 ]
 maintainers = [
-  {name = "Umberto Zerbinati", email = "umberto.zerbinati@maths.ox.ac.uk"},
+  "Umberto Zerbinati <umberto.zerbinati@maths.ox.ac.uk>",
 ]
-license = {file = "LICENSE.txt"}
-dependencies = [
-  "mpi4py",
-  "numpy",
-  "scipy",
+readme = "README.md"
+licence = "MIT"
+packages = [
+    {include = "ngsPETSc"}
 ]
-requires-python = ">=3.8"
+documentation = "https://ngspetsc.readthedocs.io/en/latest/"
+repository = "https://github.com/NGSolve/ngsPETSc"
 classifiers = [
   "Programming Language :: Python :: 3",
   "License :: OSI Approved :: MIT License",
   "Operating System :: OS Independent",
   "Development Status :: 3 - Alpha",
 ]
 
-[project.urls]
-Documentation = "https://ngspetsc.readthedocs.io/en/latest/"
-Repository = "https://github.com/NGSolve/ngsPETSc"
+[tool.poetry.dependencies]
+python = "^3.9"
+netgen-mesher = "^6.2"
+netgen-occt = "^7.8"
+petsc4py =  "^3.22.1"
+mpi4py = "^4"
+numpy = "^2.0"
+scipy = "^1"
```

Comment:
Are these explicit dependencies only needed to build the wheel or they are also used to install the package via pip? If the latter case, then they seem too strict

Reply:
They probably are a bit strict, but at least initially I want to get something working. The main aim of this PR (from Firedrake's viewpoint) is to use versioned ngsPETSc for the CI.
Firedrake requires python >=3.9 (and soon >=3.10) and petsc4py==3.22.1 and numpy >= 2.0.

Of course it's up to ngsPETSc which versions of dependencies it wasn't to support!

Reply:
I don't use firedrake with ngsPETSc, and we should find a way to relax those requirements here. If you need these specific requirements for firedrake CI you can do it there, not here, unless I'm missing something.

Reply:
I think we're agreeing with each other. Poetry by default adds a very constrained version range, these values (except for petsc4py, which is more strict and scipy which is less strict) have been edited to be pinned to minor releases, we can probably relax to major releases for most packages. I don't know enough about netgen-mesher and netgen-occt to comment further on those constraints though.